File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    
    const stack = [];
    
    let count = 0;
    
    for(let i = 0; i < s.length; i++){
        const char = s[i];
        
        // 스택의 마지막 문자과 문자가 동일하면 stack에서 제외 
        // 동일하면 stack에 추가 
        if(stack[stack.length - 1] && stack[stack.length - 1] !== char){
            stack.pop();
        } else {
            stack.push(char);
        }
        
        if(stack.length === 0 || (stack.length > 0 && i === s.length - 1)){
            count++;
        }
    }
    
    
    return count;
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    count = 0
    
    curIndex = 0
    same = 0
    diff = 0
    for i in range(len(s)):
        if curIndex != i:
            if s[curIndex] == s[i]:
                same += 1
            else:
                diff += 1
                
            if same == diff:
                curIndex = i + 1
                count += 1
                same = 0
                diff = 0
        else:
            same += 1
    
    if same != 0 or diff != 0:
        count += 1

    return count
        
        
                